cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -227,18 +227,20 @@ public RepositoryAppsResponse list(NamespaceId namespace, RepositoryConfig repoC
       // TODO CDAP-20420 Implement bulk fetch file hash in RepositoryManager
       for (File configFile : configFiles) {
         String applicationName = FileUtils.getNameWithoutExtension(configFile.getName());
+        Path appConfigRelativePath = repositoryManager.getFileRelativePath(applicationName);
```

Comment:
We need the `.json` extension here for the filename

Reply:
Let's also add a test for this

Reply:
Yeah it should be `configFile.getName()`. Fixed.
The path prefix is already covered in remote-runner tests. The inmemory-runner tests mocks the getFileRelativePath but it also verifies that the correct input is provided